react-forms/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,30 @@
+import { ChangeEvent, useState } from 'react';
+
+export function ControlledForm() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  let state;
+  const handleChangeUsername = (event: ChangeEvent<HTMLInputElement>) => {
+    setUsername(event.target.value);
+  };
+  const handleChangePassword = (event: ChangeEvent<HTMLInputElement>) => {
+    setPassword(event.target.value);
+  };
+  const handleSubmit = () => {
+    state = { username, password };
+    console.log('State:', state);
+  };
+
+  return (
+    <>
```

Comment:
You should put a form element around these inputs and handle the onSubmit event of the form element. What you've done works but it doesn't give a good user experience with accessibility, etc., since there is no form element.